packages/relay-compiler/generic/core/RelayWatchmanClient.js

```diff
@@ -17,6 +17,15 @@ const watchman = require('fb-watchman');
 class RelayWatchmanClient {
   _client: any;
 
+  static createIfAvailable(): Promise<?RelayWatchmanClient> {
+    return new Promise((resolve) => {
+      const client = new RelayWatchmanClient();
+      client.on('error', () => resolve(null));
+      client.hasCapability('relative_root')
+        .then(() => resolve(client));
```

Comment:
Right now this is always resolving to the client, regardless of the return value of relative_root - is that intended? Do you mean for this to be

```js
.then(hasRelativeRoot => hasRelativeRoot ? client : null)
```